vignettes/caseStudy3-EnvironmentalComplexity.Rmd

```diff
@@ -0,0 +1,241 @@
+    fig_height: 7
+    fig_width: 10
+    toc: yes
+    toc_depth: 2
+    number_sections: true
+vignette: >
+  %\VignetteIndexEntry{caseStudy3-EnvironmentalComplexity}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+  \usepackage[utf8]{inputenc}
+---
+
+```{r, echo=FALSE}
+knitr::opts_chunk$set(
+  cache = FALSE,
+  fig.width = 9,
+  message = FALSE,
+  warning = FALSE)
+```
+
+# Introduction
+
+`miaSim` implements tools for microbiome data simulation based on different ecological modeling assumptions. These can be used to simulate species abundance matrices, including time series. For a detailed function documentation, see the [function reference page](https://microbiome.github.io/miaSim/reference/index.html)
+
+Install the Bioconductor release version with
+
+```{r install-bioc, eval=FALSE}
+if (!requireNamespace("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+```
+
+Load the library
+
+```{r load, eval=TRUE}
+library(miaSim)
+```
+
+# Case study 3: Environmental complexity impacts on different communities
+
+(Reference: [Non-additive microbial community responses to environmental complexity](https://doi.org/10.1038/s41467-021-22426-3))
+
+The aim of this case study is to recalculate the patterns shown in the existing publication in the above reference. More specifically, the left part of this Figure 2. [![Figure 2](https://media.springernature.com/full/springer-static/image/art%253A10.1038%252Fs41467-021-22426-3/MediaObjects/41467_2021_22426_Fig2_HTML.png?as=webp)](https://www.nature.com/articles/s41467-021-22426-3#Fig2)
+
+In this case study, consumer-resource model was used. The number of species in 3 distinct communities varied so that communities are named after the number of species (i.e. com13, com3, and com4).
+
+For each community, a floor of different numbers of carbon resources (nutrients) is set from 1, 2, 4, 8, 16, to 32.
+
+In the original experiment, value of OD600 is selected as y-axis to represent the growth yield. In our simulations, however, the total number of organisms(number of all individuals) is used to reflect the growth yield.
+
+## Load dependencies
+
+```{r}
+
+library(ggplot2)
+library(miaSim)
+library(vegan)
+library(ggplot2)
+library(reshape2)
+```
+
+## Set random seed
+
+```{r}
+set.seed(42)
+```
+
+## Set number of repeats
+
+```{r}
+n_rep <- 50
+```
+
+## initial the output dataframes to store data
+
+```{r}
+
+result_df <- data.frame(
+    n_species = integer(),
+    theta = numeric(),
+    i = integer(),
+    n_resources = integer(),
+    value = numeric()
+)
+result_df2 <- data.frame(
+  matrix(NA, nrow = 0, ncol = 13, dimnames = list(NULL, paste0("sp", seq_len(13))))
+)
+sorensen_df <- data.frame(
+    n_species = integer(),
+    theta = numeric(),
+    rho_mean = numeric(),
+    rho_sd = numeric()
+)
+
+```
+
+## generating function
+
+this function generates a data frame, where each row is arranged in an increasing dissimilarity to the first row.
```

Comment:
this is shared with another vignette page; would it make sense to just add this in miaSim package in order to simplify the vignette example? This part is taking now perhaps catching attention from other more central aspects?